.github/workflows/ci.yml

```
@@ -177,6 +177,126 @@ jobs:
       - name: Find unused translation strings
         run: ./stuff/unused_translation_strings.py
 
+  cypress:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Install wait-for-it
+        run: |
+          sudo apt-get update -y && sudo apt-get install -y wait-for-it
+
+      - name: Download the Docker containers
+        run: |
+          docker-compose pull
+
+      - name: Start the Docker containers
+        run: |
+          docker-compose up --no-build --detach
+
+          # Add the container names as aliases to localhost
+          cat << EOF | sudo tee --append /etc/hosts
+          127.0.0.1 mysql grader runner gitserver broadcaster
+          EOF
+
+          # Wait for some of the containers
+          wait-for-it -t 30 mysql:13306
+          wait-for-it -t 30 gitserver:33861
+
+      - name: Create default configuration files
+        run: |
+          cat > frontend/server/config.php <<EOF
+          <?php
+          define('OMEGAUP_ALLOW_PRIVILEGE_SELF_ASSIGNMENT', true);
+          define('OMEGAUP_CSP_LOG_FILE', '/tmp/csp.log');
+          define('OMEGAUP_DB_HOST', 'mysql');
+          define('OMEGAUP_DB_NAME', 'omegaup');
+          define('OMEGAUP_DB_PASS', 'omegaup');
+          define('OMEGAUP_DB_USER', 'omegaup');
+          define('OMEGAUP_ENVIRONMENT', 'development');
+          define('OMEGAUP_LOG_FILE', '/tmp/omegaup.log');
+          define('OMEGAUP_ENABLE_SOCIAL_MEDIA_RESOURCES', false);
+          define('OMEGAUP_GITSERVER_URL', 'http://gitserver:33861');
+          define('OMEGAUP_GRADER_URL', 'https://grader:21680');
+          define('OMEGAUP_GITSERVER_SECRET_TOKEN', 'secret');
+          define('REDIS_HOST', 'localhost');
+          define('REDIS_PASS', '');
+          define('SMARTY_CACHE_DIR', '/tmp');
+          define('OMEGAUP_CSRF_HOSTS', ['frontend', '127.0.0.1']);
+          EOF
+
+          cat > ~/.my.cnf << EOF
+          [client]
+          port=13306
+          host=mysql
+          protocol=tcp
+          user=root
+          password=omegaup
+          EOF
+
+      - name: Install Python dependencies
+        run: |
+          python3 -m pip install --user setuptools
+          python3 -m pip install --user wheel
+          python3 -m pip install --user --upgrade urllib3
+          python3 -m pip install --user \
+            mysqlclient==2.0.1 \
+            pytest \
+            pytest-xdist \
+            flaky
```

```suggestion
            -r stuff/docker/requirements.txt \
            mysqlclient==2.0.1
```

esos módulos sólo se usan para selenium